executor/executor.go

```diff
@@ -691,9 +691,19 @@ func (e *ShowDDLJobQueriesExec) Next(ctx context.Context, req *chunk.Chunk) erro
 	}
 	numCurBatch := mathutil.Min(req.Capacity(), len(e.jobs)-e.cursor)
 	for _, id := range e.jobIDs {
-		for i := e.cursor; i < e.cursor+numCurBatch; i++ {
-			if id == e.jobs[i].ID {
+		i := e.cursor
+		numOfRowsToWrite := numCurBatch
+		appendedJobID := make(map[int64]struct{})
+		for {
+			_, ok := appendedJobID[e.jobs[i].ID]
+			if id == e.jobs[i].ID && !ok {
+				appendedJobID[e.jobs[i].ID] = struct{}{}
 				req.AppendString(0, e.jobs[i].Query)
+				numOfRowsToWrite--
+			}
+			i++
+			if i >= len(e.jobs) || numOfRowsToWrite == 0 {
```

Comment:
`numCurBatch` does not have same meaning as in `ShowDDLJobQueriesWithRangeExec`, it's a batch size of cursor loop, not the amount we should write in this `Next` invocation. If we breaks here, e.cursor will be forwarded while some matched rows are not output